src/binary/merkle_tree.rs

```diff
@@ -35,9 +35,19 @@ pub struct MerkleTree<StorageType> {
     leaves_count: u64,
 }
 
+/// Flat array of merkle's tree nodes.
+pub struct NodesArray;
```

Comment:
Maybe it would be more appropriate to call this a `NodesIndex`, since the purpose is to correlate u64's with Node's? It seems a little presumptuous to assume it's an array.

Reply:
Yeah, it is wrong naming, `MerkleTree` works with it as with array, but the storage can be anything=)
`NodesStorage` or `NodesTable` should be better. The comment will describe that the key is the `u64` index.